File: test-junit5/src/test/java/io/micronaut/test/junit5/DisableEmbeddedApplicationTest.java

package io.micronaut.test.junit5;

import io.micronaut.runtime.EmbeddedApplication;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import org.junit.jupiter.api.MethodOrderer;
import org.junit.jupiter.api.Order;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.TestMethodOrder;

import javax.inject.Inject;

import static org.junit.Assert.assertFalse;

@MicronautTest(startApplication = false, rebuildContext = true)
@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
class DisableEmbeddedApplicationTest {

    @Inject
    private EmbeddedApplication<?> embeddedApplication;

    @Test
    @Order(1)
    void embeddedApplicationIsNotStartedWhenContextIsStarted() {
        assertFalse(embeddedApplication.isRunning());
    }

    @Test
    @Order(2)
    void embeddedApplicationIsNotStartedWhenContextIsRebuilt() {
        assertFalse(embeddedApplication.isRunning());
    }
}

File: test-kotest/src/test/kotlin/io/micronaut/test/kotest/DisableEmbeddedApplicationTest.kt
package io.micronaut.test.kotest

import io.kotest.core.spec.style.StringSpec
import io.kotest.matchers.shouldBe
import io.micronaut.runtime.EmbeddedApplication
import io.micronaut.test.extensions.kotest.annotation.MicronautTest
import javax.inject.Inject

@MicronautTest(startApplication = false, rebuildContext = true)
internal class DisableEmbeddedApplicationTest: StringSpec() {

    @Inject lateinit var embeddedApplication: EmbeddedApplication<*>

    init {
        "test embedded server is not started"() {
            embeddedApplication.isRunning shouldBe false
        }

        "test embedded server is not started after context rebuild"() {
            embeddedApplication.isRunning shouldBe false
        }
    }
}
